BungieFrameWork/BFW_Source/BFW_LocalInput/Platform_Win32/BFW_LI_Platform_Win32.c

```diff
@@ -1,3 +1,4 @@
+#if UUmPlatform == UUmPlatform_Win32
```

Comment:
Rather than adding these guards to every platform-specific file, it would be better to simply exclude the from the compilation if it's not applicable for that platform. This is pretty straight forward via CMake.

Also, BFW.h defines UUmPlatform, but won't get included here because UUmPlatform hasn't been set yet (so even on Windows, UUmPlatform isn't UUmPlatform_Win32.) So it's a bit odd. I think it's better practice to just avoid this.

Reply:
It's included via PCH.
An advantage of this is that the resulting project file would contain all files, even if not relevant to the target. Some files already had this setup as such as well.

As you said, directly excluding them from the generated project file is also possible. That was in fact my first approach until I saw that mechanism of checking `UUmPlatform`.
Doing that would work as well though, but let me know if maybe having all files available in the generated project file is a good argument to keep it this way.

Reply:
You're absolutely right, I'd totally forgotten haha. That said, I'd still prefer to avoid this and instead just exclude the files from the compilation if they're not valid for that platform.